SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityDeployChecker.java

```diff
@@ -265,12 +268,25 @@ private void finishDeploy(SingularityRequestWithState requestWithState, Optional
       cleanupTasks(pendingDeploy, request, deployResult, tasksToKill);
     }
 
-    if (!request.isDeployable() && !request.isOneOff()) {
-      // TODO should this override? What if someone has mucked with the pending queue for this deploy ?
-      requestManager.addToPendingQueue(new SingularityPendingRequest(request.getId(), pendingDeploy.getDeployMarker().getDeployId(), deployResult.getTimestamp(),
-        pendingDeploy.getDeployMarker().getUser(), deployResult.getDeployState() == DeployState.CANCELED ? PendingType.DEPLOY_CANCELLED : PendingType.NEW_DEPLOY,
-        deploy.isPresent() ? deploy.get().getSkipHealthchecksOnDeploy() : Optional.<Boolean> absent(), pendingDeploy.getDeployMarker().getMessage()));
+    if (deploy.isPresent() && deploy.get().getRunImmediately().isPresent()) {
+      Optional<SingularityPendingRequest> maybePendingRequest = buildPendingRequest(request,
+          pendingDeploy,
+          deployResult,
+          deploy.get().getRunImmediately());
+      if (maybePendingRequest.isPresent()) {
```

Comment:
for this and the one below. We should always make sure _some_ pending request gets enqueued for a `!deployable && !oneOff` case. If it doesn't get enqueued it's possible we could miss scheduling the next run of a cron or something like that.